RoddyCore/src/de/dkfz/roddy/execution/io/fs/FileSystemAccessProvider.groovy

```diff
@@ -417,167 +423,168 @@ public class FileSystemAccessProvider {
             }
 
-            return _groupIDsByGroup[groupID];
+            return _groupIDsByGroup[groupID]
         }
     }
 
     boolean isGroupAvailable(String groupID) {
-        getListOfGroups().find { it == groupID }
+        listOfGroups.find { it == groupID }
     }
 
     private String _getOwnerOfPath(File file) {
-        String cmd = commandSet.getOwnerOfPathCommand(file);
-        ExecutionResult er = ExecutionService.getInstance().execute(cmd);
-        return er.resultLines[0];
+        String cmd = commandSet.getOwnerOfPathCommand(file)
+        ExecutionResult er = ExecutionService.instance.execute(cmd)
+        return er.resultLines[0]
     }
 
     /**
      * Returns the creator / current owner of a path.
      */
-    public String getOwnerOfPath(File file) {
-        if (ExecutionService.getInstance().canQueryFileAttributes()) {
-            FileAttributes attributes = ExecutionService.getInstance().queryFileAttributes(file);
+    String getOwnerOfPath(File file) {
+        if (ExecutionService.instance.canQueryFileAttributes()) {
+            FileAttributes attributes = ExecutionService.instance.queryFileAttributes(file)
             if (attributes == null || attributes.userID == null)
-                return _getOwnerOfPath(file);
+                return _getOwnerOfPath(file)
             if (attributes.userID.isInteger()) {
                 if (!uidToUserCache.containsKey(attributes.userID)) {
-                    uidToUserCache[attributes.userID] = _getOwnerOfPath(file);
+                    uidToUserCache[attributes.userID] = _getOwnerOfPath(file)
                 }
                 return uidToUserCache[attributes.userID]
             }
-            return attributes.userID;
+            return attributes.userID
         } else
-            return _getOwnerOfPath(file);
+            return _getOwnerOfPath(file)
     }
 
 
-    public boolean writeTextFile(File file, String text, ExecutionContext context) {
+    boolean writeTextFile(File file, String text, ExecutionContext context) {
         return writeTextFile(file, text) &&
-                setDefaultAccessRights(file, context);
+                setDefaultAccessRights(file, context)
     }
 
-    public boolean writeTextFile(File file, String text) {
-        ExecutionService eService = ExecutionService.getInstance()
-        if (eService.canWriteFiles()) { //Let the execution service do this.
-            return eService.writeTextFile(file, text);
+    boolean writeTextFile(File file, String text) {
+        ExecutionService eService = ExecutionService.instance
+        if (eService.canWriteFiles()) {
+            return eService.writeTextFile(file, text)
         } else if (eService.isLocalService()) {
             if (!file.getParentFile().exists())
-                file.getParentFile().mkdirs();
-            file.write(text);
-            return true;
+                file.getParentFile().mkdirs()
+            file.write(text)
```

Comment:
The indentation is because of the if-statement before. Indented is only the single-line if-branch. The next line is outside the if-statement. While this is correct Groovy (and Java) syntax, I find it unclear. But it is found throughout the Roddy code, and I just changed the ';'.